sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -299,6 +300,29 @@ public Instant getCurrentTimestamp() throws NoSuchElementException {
     return curTimestamp;
   }
 
+  @Override
+  public byte[] getCurrentRecordId() throws NoSuchElementException {
```

Comment:
can you improve unit test to cover the new behavior?

Reply:
Added some testing, but please lmk if you have suggestions for more approaches to test this behavior.